alembic/versions/2024_09_11_18dbadd8c436_rename_fastq_to_raw_data.py

```diff
@@ -0,0 +1,169 @@
+        session.query(Analysis).filter(Analysis.workflow == "fastq").update(
+            {"workflow": Workflow.RAW_DATA}, synchronize_session="evaluate"
+        )
+        op.alter_column("analysis", "workflow", type_=new_enum, existing_nullable=True)
+        # Order
+        op.alter_column("order", "workflow", type_=mysql.VARCHAR(64), existing_nullable=True)
+        session.query(Order).filter(Order.workflow == "fastq").update(
+            {"workflow": Workflow.RAW_DATA}, synchronize_session="evaluate"
+        )
+        op.alter_column("order", "workflow", type_=new_enum, existing_nullable=True)
+        # Application Limitation
+        op.alter_column(
+            "application_limitations", "workflow", type_=mysql.VARCHAR(64), existing_nullable=True
+        )
+        session.query(ApplicationLimitations).filter(
+            ApplicationLimitations.workflow == "fastq"
+        ).update({"workflow": Workflow.RAW_DATA}, synchronize_session="evaluate")
+        op.alter_column(
+            "application_limitations", "workflow", type_=new_enum, existing_nullable=True
+        )
+        session.commit()
+    finally:
+        session.close()
+
+
+def downgrade():
+    bind = op.get_bind()
+    session = orm.Session(bind=bind)
+    try:
+        # Case
+        op.alter_column("case", "data_analysis", type_=mysql.VARCHAR(64), existing_nullable=True)
+        session.query(Case).filter(Case.data_analysis == Workflow.RAW_DATA).update(
+            {"data_analysis": "fastq"}, synchronize_session="evaluate"
+        )
+        op.alter_column("case", "data_analysis", type_=old_enum, existing_nullable=True)
+        # Analysis
+        op.alter_column("analysis", "workflow", type_=mysql.VARCHAR(64), existing_nullable=True)
+        session.query(Analysis).filter(Analysis.workflow == Workflow.RAW_DATA).update(
+            {"workflow": "fastq"}, synchronize_session="evaluate"
+        )
+        op.alter_column("analysis", "workflow", type_=old_enum, existing_nullable=True)
+        # Order
+        op.alter_column("order", "workflow", type_=mysql.VARCHAR(64), existing_nullable=True)
+        session.query(Order).filter(Order.workflow == Workflow.RAW_DATA).update(
+            {"workflow": "fastq"}, synchronize_session="evaluate"
+        )
+        op.alter_column("order", "workflow", type_=old_enum, existing_nullable=True)
+        # Application Limitation
+        op.alter_column(
+            "application_limitations", "workflow", type_=mysql.VARCHAR(64), existing_nullable=True
+        )
+        session.query(ApplicationLimitations).filter(
+            ApplicationLimitations.workflow == Workflow.RAW_DATA
+        ).update({"workflow": "fastq"}, synchronize_session="evaluate")
+        op.alter_column(
+            "application_limitations", "workflow", type_=old_enum, existing_nullable=True
+        )
+        session.commit()
+    finally:
+        session.close()
```

Comment:
once we have pacbio samples in here, this downgrade would change their workflow to fastq as well, right? Just out of curiosity. Maybe we can add an extra check where data delivery is also fastq? When we do the query?

Reply:
Good spotting! That would still break though since the pacbio raw-data entries would not be allowed with the old enum. Don't know what the best approach here is.

Reply:
Also, there are some cases with `data_analysis=="fastq"` but `data_delivery=="analysis"` and `data_delivery=="scout"`, which will cause an incomplete migration.

Reply:
We will leave it as it is